posts/Yoann/2017-02-22-hibernate-search-5-7-0-Final-and-5-6-1-Final.adoc

```diff
@@ -0,0 +1,118 @@
+= Hibernate Search 5.7.0.Final release is out! (and, as a bonus, 5.6.1)
+Yoann Rodière
+:awestruct-tags: [ "Hibernate Search", "Elasticsearch", "Releases" ]
+:awestruct-layout: blog-post
+---
+
+You asked for it, we deliver: Hibernate Search 5.7.0.Final, with support for Hibernate ORM 5.2.8, is out!
+And of course, like previous versions, it features experimental support for Elasticsearch integration
+(even with some improvements).
+
+Stuck on an older ORM versions? Don't worry, we also released a bugfix version for 5.6, namely 5.6.1.Final.
+
+[WARNING]
+====
+Hibernate Search 5.7.0.Final is only compatible with Hibernate ORM 5.2.3 and later. There isn't any version of Hibernate Search compatible Hibernate ORM 5.2.0 to 5.2.2.
+
+If you really need to use Hibernate ORM 5.0.x or 5.1.x and cannot upgrade,
+please use Hibernate Search 5.6.1.Final.
+====
+
+== What's new on 5.7 since the candidate release?
+
+Below are the main changes since the candidate release.
+
+For a full list of changes since 5.7.0.CR1, please see https://hibernate.atlassian.net/secure/ReleaseNote.jspa?projectId=10061&version=27300[the release notes].
+For a full list of changes since 5.6.0.Final, please see https://hibernate.atlassian.net/issues/?jql=fixVersion%20in%20(5.7.0.Alpha1%2C%205.7.0.Beta1%2C%205.7.0.Beta2%2C%205.7.0.CR1%2C%205.7.0.Final)%20AND%20fixVersion%20not%20in%20(5.6.0.Alpha1%2C%205.6.0.Alpha2%2C%205.6.0.Alpha3%2C%205.6.0.Beta1%2C%205.6.0.Beta2%2C%205.6.0.Beta3%2C%205.6.0.Beta4%2C%205.6.0.CR1%2C%205.6.0.Final)[this list of tickets on our JIRA instance].
+
+=== Common changes
+
+ * https://hibernate.atlassian.net/browse/HSEARCH-2574[HSEARCH-2574]: when it makes sense,
+   you can now repeat Hibernate Search annotations on a single attribute/method.
+   For example, if you want to add two index fields on the same `title` attribute, you can write:
++
+====
+[source, Java]
+----
+    @Field(analyzer = @Analyzer(definition = "myAnalyzer")
+    @Field(name = "title_sort", analyzer = @Analyzer(definition = "myAnalyzerForSort")
+    private String title;
+====
++
+Thus you don't have to use the weird composite annotations (`@Fields`) anymore.
+But of course, those are still supported.
+ * https://hibernate.atlassian.net/browse/HSEARCH-2588[HSEARCH-2588]: some methods in the Sort DSL allowed to specify
+   the sort field type explicitly, which was useful when sorting on fields added through custom field bridges.
+   We improved support for `MetadataProvidingFieldBridge`,
+   so that you now only need to implement this interface on your custom field bridges,
+   and won't need anything special when using the sort DSL anymore.
+   As a result, the `byField(String, SortField.Type)` and `andByField(String, SortField.Type)` methods
+   in the sort DSL have been deprecated.  
+ * https://hibernate.atlassian.net/browse/HSEARCH-2587[HSEARCH-2587]: an NPE could occur when using the Sort DSL to sort on embedded (`@IndexedEmbedded`) fields; this is now fixed.
```

Comment:
I would write **a** NPE but we can probably discuss it. Just mentioning it to be safe.